src/wallet/rpcimportmulti.cpp

```diff
@@ -0,0 +1,207 @@
+#include "sync.h"
+#include "util.h"
+#include "utiltime.h"
+#include "wallet.h"
+
+#include <fstream>
+#include <stdint.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+#include <univalue.h>
+
+#include <boost/foreach.hpp>
+
+
+//namespaces
+using namespace std;
+
+//extern functions
+extern void ImportAddress(const CBitcoinAddress& address, const string& strLabel);
+extern void ImportScript(const CScript& script, const string& strLabel, bool isRedeemScript);
+extern bool EnsureWalletIsAvailable(bool avoidException);
+extern void EnsureWalletIsUnlocked();
+
+//binding names
+const string TYPE_PARAM = "type";
+const string TYPE_PRIVKEY_PARAM = "privkey";
+const string TYPE_PUBKEY_PARAM = "pubkey";
+const string TYPE_ADDRESS_PARAM = "address";
+const string VALUE_PARAM = "value";
+const string LABEL_PARAM = "label";
+const string TIMESTAMP_PARAM = "timestamp";
+const string P2SH_PARAM = "p2sh";
+
+void ImportAddressKey(const string& script,const string& label, bool isF2SH){
+    CBitcoinAddress address(script);
+    if (address.IsValid()) {
+        if (isF2SH)
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Cannot use the p2sh flag with an address - use a script instead");
+        ImportAddress(address, label);
+    } else if (IsHex(script)) {
+        std::vector<unsigned char> data(ParseHex(script));
+        ImportScript(CScript(data.begin(), data.end()), label, isF2SH);
+    } else {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid Bitcoin address or script");
+    }
+}
+
+void ImportPublicKey(const string& pubkey,const string& label){
+    //Legacy code - begin
+    if (!IsHex(pubkey))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Pubkey must be a hex string");
+    std::vector<unsigned char> data(ParseHex(pubkey));
+    CPubKey pubKey(data.begin(), data.end());
+    if (!pubKey.IsFullyValid())
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Pubkey is not a valid public key");
+
+    ImportAddress(CBitcoinAddress(pubKey.GetID()), label);
+    ImportScript(GetScriptForRawPubKey(pubKey), label, false);
+    //Legacy code - end
+}
+
+void ImportPrivateKey(const string& privkey,const string& label){
+    //Legacy code - begin
+    CBitcoinSecret vchSecret;
+    bool fGood = vchSecret.SetString(privkey);
+
+    if (!fGood) throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid private key encoding");
+
+    CKey key = vchSecret.GetKey();
+    if (!key.IsValid()) throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Private key outside allowed range");
+
+    CPubKey pubkey = key.GetPubKey();
+    assert(key.VerifyPubKey(pubkey));
+    CKeyID vchAddress = pubkey.GetID();
+    {
+        pwalletMain->MarkDirty();
+        pwalletMain->SetAddressBook(vchAddress, label, "receive");
+
+        // Don't throw error in case a key is already there
+        if (pwalletMain->HaveKey(vchAddress))
+            return ;//NullUniValue;
```

Comment:
@promag i'll remove the comment. When the wallet already has the key should it return true or false on the response?